src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -62,16 +62,42 @@ public void storeMessage(Message message) {
             .setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient))
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    messages = message(user);
+    }
+
+    return messages;
+  }
+  public List<Message> getAllMessages(){
+    List<Message> messages = new ArrayList<>();
+   
+    Query query = new Query("Message")
+      .addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
The refactoring is close but not quite it. Notice how the Query is different between the two methods. I know this is someone else's code and refactoring can be daunting. Your project is the only one that asked for refactoring, sorry about that. Feel free to give it another try and I'll be happy to help out with it, but if you prefer to keep the original methods from project instructions, that's also alright.

Reply:
I'll give it another try! Could I make results a parameter for the helper method since that is what the loop iterates through?

Reply:
Yes, `results` would be a good choice for the parameter for the helper method. Alternatively, you can go a step before that and make `query` a parameter. That would allow you to put the `PreparedQuery results = datastore.prepare(query);` line inside the helper method too.

Reply:
Awesome! Thank you for the help :) I committed the changes and think they changed on this pull request, but let me know.

Reply:
Commit worked! I can see the changes. Great progress! Don't forget to test your code by running `mvn appengine:devserver` which will show you all compile errors.